java/piranha/src/test/resources/com/uber/piranha/XPMethodChainCasesTreatmentDoNotAllowMatchingArgMethodInvc.java

```diff
@@ -0,0 +1,74 @@
+package com.uber.piranha;
+
+class XPMethodChainCases {
+
+  interface SomeParam {
+    @BoolParam(key = "other_flag")
+    BoolParameter otherFlag();
+
+    static SomeParam create(Parameter cp) {
+      return null;
+    }
+  }
+
+  public BoolParameter staleFlag() {
+    return null;
+  }
+
+  public void testDontMatchNonInstanceNested() {
+    // Does not Match
+    if (staleFlag().getValue()) {
```

Comment:
Shouldn't this be defined as part of `SomeParam` to be matched in the first place?

Reply:
This is just a negative test case. It makes sure that if there is some other instance method with similar name, we dont update it.